### ISSUE TITLE
Nested Resource Not Working

### ISSUE THREAD

**Issue by nahtnam** [SeyZ/jsonapi-serializer#48]
Hey!

I am trying to get nested serializers to work but it refuses to add the nested attributes.

Here is what im doing:

Fetching the status with its related datacenters:
```
Status.where('id', 1).fetch({
      withRelated: 'datacenters'
    }).then((status) => {
        res.json(Serializer.serialize(status));
    });
```
which sends this object
```
{
    id: 1,
    login: 'normal',
    community: 'delayed',
    matchmaking: 'normal',
    online_servers: 192897,
    online_players: 665583,
    searching: 7280,
    wait_time: 39,
    created_at: SunNov22201510: 38: 01GMT-0800(PST),
    updated_at: SunNov22201510: 38: 01GMT-0800(PST),
    datacenters: [
        {
            id: 1,
            name: 'EUWest',
            capacity: 'full',
            load: 'medium',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 2,
            name: 'EUEast',
            capacity: 'full',
            load: 'medium',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 3,
            name: 'USSouthwest',
            capacity: 'full',
            load: 'low',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 4,
            name: 'USSoutheast',
            capacity: 'full',
            load: 'idle',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 5,
            name: 'India',
            capacity: 'offline',
            load: 'full',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 6,
            name: 'EUNorth',
            capacity: 'full',
            load: 'medium',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 7,
            name: 'Emirates',
            capacity: 'full',
            load: 'idle',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 8,
            name: 'USNorthwest',
            capacity: 'full',
            load: 'idle',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 9,
            name: 'SouthAfrica',
            capacity: 'high',
            load: 'high',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 10,
            name: 'Brazil',
            capacity: 'full',
            load: 'high',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 11,
            name: 'USNortheast',
            capacity: 'full',
            load: 'low',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 12,
            name: 'Japan',
            capacity: 'full',
            load: 'idle',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 13,
            name: 'Singapore',
            capacity: 'full',
            load: 'low',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        },
        {
            id: 14,
            name: 'Australia',
            capacity: 'full',
            load: 'idle',
            steam_status_id: 1,
            created_at: SunNov22201510: 38: 01GMT-0800(PST),
            updated_at: SunNov22201510: 38: 01GMT-0800(PST)
        }
    ]
}
```

and then is serialized like so:

```
serialize: (data) => {
    const json = data.toJSON()
    const status = new JSONAPISerializer('status', json, {
      topLevelLinks: { self: `${ENV.host}/steam/status` },
      pluralizeType: false,
      attributes: ['id', 'login', 'community', 'matchmaking', 'online_servers', 'online_players', 'searching', 'wait_time', 'updated_at'],
      datacenters: {
        attributes: ['id', 'name', 'load']
      }
    });
    return status;
  }
```

That code is returning this json:
```
{
  "links": {
    "self": "http://localhost:3000/steam/status"
  },
  "data": {
    "type": "status",
    "id": "1",
    "attributes": {
      "id": 1,
      "login": "normal",
      "community": "delayed",
      "matchmaking": "normal",
      "online-servers": 192897,
      "online-players": 665583,
      "searching": 7280,
      "wait-time": 39,
      "updated-at": "2015-11-22T18:38:01.417Z"
    }
  }
}
```

As you can see the nested stuff isnt attached. Any help would be appreciated.

Thanks.

**Comment by SeyZ** [comment]
Hey @nahtnam I think you just missed to add the `datacenters` attribute to the `attributes` array. The following code should work, right?

```
serialize: (data) => {
    const json = data.toJSON()
    const status = new JSONAPISerializer('status', json, {
      topLevelLinks: { self: `${ENV.host}/steam/status` },
      pluralizeType: false,
      attributes: ['id', 'login', 'community', 'matchmaking', 'online_servers', 'online_players', 'searching', 'wait_time', 'updated_at', 'datacenters'],
      datacenters: {
        attributes: ['id', 'name', 'load']
      }
    });
    return status;
  }
```

**Comment by nahtnam** [comment]
Ah yes.

It works now.

Thanks @SeyZ!

**Comment by nahtnam** [comment]
Oh, one more question.

In the above context and this code:
```
serialize: (data) => {
    const json = data.toJSON()
    const status = new JSONAPISerializer('status', json, {
      topLevelLinks: { self: `${ENV.host}/steam/status` },
      pluralizeType: false,
      attributes: ['id', 'login', 'community', 'matchmaking', 'online_servers', 'online_players', 'searching', 'wait_time', 'updated_at', 'datacenters'],
      datacenters: {
        attributes: ['id', 'name', 'capacity', 'load']
      }
    });
    return status;
  }
```
it doesnt assign the correct attributes:
```
{
  "links": {
    "self": "http://localhost:3000/steam/status"
  },
  "data": {
    "type": "status",
    "id": "1",
    "attributes": {
      "id": 1,
      "login": "normal",
      "community": "delayed",
      "matchmaking": "normal",
      "online-servers": 189034,
      "online-players": 627445,
      "searching": 6149,
      "wait-time": 37,
      "updated-at": "2015-11-22T19:35:01.888Z",
      "datacenters": [
        {
          "id": 1,
          "name": "EU West",
          "capacity": "full",
          "load": "medium",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 2,
          "name": "EU East",
          "capacity": "full",
          "load": "medium",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 3,
          "name": "US Southwest",
          "capacity": "full",
          "load": "low",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 4,
          "name": "US Southeast",
          "capacity": "full",
          "load": "idle",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 5,
          "name": "India",
          "capacity": "offline",
          "load": "full",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 6,
          "name": "EU North",
          "capacity": "full",
          "load": "medium",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 7,
          "name": "Emirates",
          "capacity": "full",
          "load": "idle",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.902Z",
          "updated-at": "2015-11-22T19:35:01.902Z"
        },
        {
          "id": 8,
          "name": "US Northwest",
          "capacity": "full",
          "load": "low",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 9,
          "name": "South Africa",
          "capacity": "high",
          "load": "high",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 10,
          "name": "Brazil",
          "capacity": "full",
          "load": "high",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 11,
          "name": "US Northeast",
          "capacity": "full",
          "load": "low",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 12,
          "name": "Japan",
          "capacity": "full",
          "load": "idle",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 13,
          "name": "Singapore",
          "capacity": "full",
          "load": "low",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        },
        {
          "id": 14,
          "name": "Australia",
          "capacity": "full",
          "load": "idle",
          "steam-status-id": 1,
          "created-at": "2015-11-22T19:35:01.903Z",
          "updated-at": "2015-11-22T19:35:01.903Z"
        }
      ]
    }
  }
}
```

As you can see, the datacenters includes things like updated-at and created-at, even though its not in the attributes.

Am I doing something wrong?

Thanks.

**Comment by nahtnam** [comment]
@SeyZ Any idea what's wrong?

Thanks.

**Comment by nahtnam** [comment]
Bump

**Comment by SeyZ** [comment]
@nahtnam Sorry for late. I'm currently working on this bug :)

**Comment by nahtnam** [comment]
Awesome, thanks.

**Comment by SeyZ** [comment]
@nahtnam I fixed your issue in the fix-48 branch. Can you test if everything is ok using this branch?
I will release it soon

Thanks for report @nahtnam :+1:

**Comment by nahtnam** [comment]
Sorry, I have finals this week. I'll try it next week.

Thanks.

**Comment by jylinman** [comment]
Hey there, just letting you know that the fix works on my end-- do you think we can move this puppy back into the master?

**Comment by SeyZ** [comment]
@jylinman Yeah sure, sorry for the delay. I will try to do that as soon as possible :)

**Comment by SeyZ** [comment]
The fix is now available on master.

**Comment by erikakers** [comment]
Was there a version bump for this fix? I'm still seeing this bug on 2.0.4 which is still the latest version.

**Comment by SeyZ** [comment]
@erikakers It's available on master. I will release a new version soon ;)
